### PR TITLE
User registration

### PR DESCRIPTION
**Pull request by osamaabdulsattar**

### CODE REVIEW

**Comment by MarwanMAlfalah** [comment]
I cannot add comments or notes to the code that you requested!

**Review by Khawlah-Mohammed — APPROVED** [review]

File: src/main/java/com/example/mytestproject/UserRegistrationRestController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    UsersDatabaseConnection connection;
+
+    public UserRegistrationRestController() {
+    }
+
+    @GetMapping("/user-registration")
+    public void registerUser(@PathVariable String username, @PathVariable String password) {
+        connection.connect();
+        connection.registerUser();
```
Comment (EngMoAmri) [inline src/main/java/com/example/mytestproject/UserRegistrationRestController.java:21]: we need a validation before registeration

File: src/main/java/com/example/mytestproject/UsersDatabaseConnection.java [hunk 2]
```diff
@@ -18,6 +18,10 @@ public void updateUser() {
         System.out.println("Updating user in the users database");
     }
 
+    public void registerUser() {
```
Comment (MarwanMAlfalah) [inline src/main/java/com/example/mytestproject/UsersDatabaseConnection.java:21]: Can you improve this fun. like this Mr. Osama:
public void registerUser(String username) {
        // Logic for registering a user in the users database
        System.out.println("User '" + username + "' registered successfully.");
    }

File: src/main/java/com/example/mytestproject/UsersDatabaseConnection.java [hunk 3]
```diff
@@ -18,6 +18,10 @@ public void updateUser() {
...
+        System.out.println("Registering user in the users database");
```
Comment (MarwanMAlfalah) [inline src/main/java/com/example/mytestproject/UsersDatabaseConnection.java:22]: And this you improve this fun. like this Mr. Osama:
public void deleteUser(String username) {
        // Logic for deleting a user from the users database
        System.out.println("User '" + username + "' deleted successfully.");
    }

File: src/main/java/com/example/mytestproject/UsersDatabaseConnection.java [hunk 4]
```diff
@@ -18,6 +18,10 @@ public void updateUser() {
...
+    }
+
     public void deleteUser() {
         System.out.println("Deleting user from the users database");
     }
```
Comment (msama1) [inline src/main/java/com/example/mytestproject/UsersDatabaseConnection.java:27]: Cascade the deletion of the user before delete the user

Comment (AbdulgaleelFareed) [inline src/main/java/com/example/mytestproject/UsersDatabaseConnection.java:27]: it very clear code

**Comment by ahdel** [comment]
sever path variable not specified

File: src/main/java/com/example/mytestproject/UsersDatabaseConnection.java [hunk 5]
```diff
@@ -18,6 +18,10 @@ public void updateUser() {
         System.out.println("Updating user in the users database");
     }
 
+    public void registerUser() {
+        System.out.println("Registering user in the users database");
+    }
+
     public void deleteUser() {
```
Comment (MarwanMAlfalah) [inline src/main/java/com/example/mytestproject/UsersDatabaseConnection.java:30]: // #You might also want to handle or propagate the exception depending on the use case

public void deleteUser() {
    try {
        // Perform the deletion operation
        deleteFromDatabase();

        // Log success message
        System.out.println("User successfully deleted from the users database");
    } catch (Exception e) {
        // Log error message
        System.err.println("Error deleting user from the users database: " + e.getMessage());
        // You might also want to handle or propagate the exception depending on the use case
    }
}

**Pull request merged**